src/blueprint/from_config.rs

```diff
@@ -349,6 +350,18 @@ impl<'a> MustachePartsValidator<'a> {
   fn new(type_of: &'a config::Type, config: &'a Config, field: &'a FieldDefinition) -> Self {
     Self { type_of, config, field }
   }
+
+  fn get_nested_type(&self, tail: &[String], type_of: &config::Type) -> Option<Type> {
+    if tail.len() == 1 {
+      return get_value_type(type_of, &tail[0]);
+    }
+
+    let head = tail[0].as_str();
+    let field = type_of.fields.get(head)?;
+    let type_of = self.config.find_type(&field.type_of)?;
+    self.get_nested_type(&tail[1..], type_of)
```

Comment:
take a look in [example](https://github.com/tailcallhq/tailcall/blob/741a1cc27e54e75132f840a68b4862fc73516bef/src/lambda/evaluation_context.rs#L125) for doing something like this without recursion

Reply:
fixed